// @aws-sdk/client-polly/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "polly",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class PollyClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultPollyHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class PollyServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, PollyServiceException.prototype);
    }
}

class LexiconNotFoundException extends PollyServiceException {
    name = "LexiconNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LexiconNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LexiconNotFoundException.prototype);
    }
}
class ServiceFailureException extends PollyServiceException {
    name = "ServiceFailureException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "ServiceFailureException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceFailureException.prototype);
    }
}
class InvalidNextTokenException extends PollyServiceException {
    name = "InvalidNextTokenException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidNextTokenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidNextTokenException.prototype);
    }
}
class EngineNotSupportedException extends PollyServiceException {
    name = "EngineNotSupportedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "EngineNotSupportedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EngineNotSupportedException.prototype);
    }
}
class InvalidTaskIdException extends PollyServiceException {
    name = "InvalidTaskIdException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTaskIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTaskIdException.prototype);
    }
}
class SynthesisTaskNotFoundException extends PollyServiceException {
    name = "SynthesisTaskNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "SynthesisTaskNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SynthesisTaskNotFoundException.prototype);
    }
}
class InvalidLexiconException extends PollyServiceException {
    name = "InvalidLexiconException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidLexiconException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLexiconException.prototype);
    }
}
class InvalidS3BucketException extends PollyServiceException {
    name = "InvalidS3BucketException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidS3BucketException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidS3BucketException.prototype);
    }
}
class InvalidS3KeyException extends PollyServiceException {
    name = "InvalidS3KeyException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidS3KeyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidS3KeyException.prototype);
    }
}
class InvalidSampleRateException extends PollyServiceException {
    name = "InvalidSampleRateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSampleRateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSampleRateException.prototype);
    }
}
class InvalidSnsTopicArnException extends PollyServiceException {
    name = "InvalidSnsTopicArnException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSnsTopicArnException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSnsTopicArnException.prototype);
    }
}
class InvalidSsmlException extends PollyServiceException {
    name = "InvalidSsmlException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSsmlException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSsmlException.prototype);
    }
}
class LanguageNotSupportedException extends PollyServiceException {
    name = "LanguageNotSupportedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LanguageNotSupportedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LanguageNotSupportedException.prototype);
    }
}
class LexiconSizeExceededException extends PollyServiceException {
    name = "LexiconSizeExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LexiconSizeExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LexiconSizeExceededException.prototype);
    }
}
class MarksNotSupportedForFormatException extends PollyServiceException {
    name = "MarksNotSupportedForFormatException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MarksNotSupportedForFormatException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MarksNotSupportedForFormatException.prototype);
    }
}
class MaxLexemeLengthExceededException extends PollyServiceException {
    name = "MaxLexemeLengthExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MaxLexemeLengthExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MaxLexemeLengthExceededException.prototype);
    }
}
class MaxLexiconsNumberExceededException extends PollyServiceException {
    name = "MaxLexiconsNumberExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MaxLexiconsNumberExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MaxLexiconsNumberExceededException.prototype);
    }
}
class UnsupportedPlsAlphabetException extends PollyServiceException {
    name = "UnsupportedPlsAlphabetException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnsupportedPlsAlphabetException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedPlsAlphabetException.prototype);
    }
}
class UnsupportedPlsLanguageException extends PollyServiceException {
    name = "UnsupportedPlsLanguageException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnsupportedPlsLanguageException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedPlsLanguageException.prototype);
    }
}
class SsmlMarksNotSupportedForTextTypeException extends PollyServiceException {
    name = "SsmlMarksNotSupportedForTextTypeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "SsmlMarksNotSupportedForTextTypeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SsmlMarksNotSupportedForTextTypeException.prototype);
    }
}
class TextLengthExceededException extends PollyServiceException {
    name = "TextLengthExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TextLengthExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TextLengthExceededException.prototype);
    }
}

const _A = "Alphabet";
const _ALC = "AdditionalLanguageCodes";
const _AS = "AudioStream";
const _At = "Attributes";
const _C = "Content";
const _CT = "CreationTime";
const _CT_ = "Content-Type";
const _CTo = "ContentType";
const _DL = "DeleteLexicon";
const _DLI = "DeleteLexiconInput";
const _DLO = "DeleteLexiconOutput";
const _DV = "DescribeVoices";
const _DVI = "DescribeVoicesInput";
const _DVO = "DescribeVoicesOutput";
const _E = "Engine";
const _ENSE = "EngineNotSupportedException";
const _G = "Gender";
const _GL = "GetLexicon";
const _GLI = "GetLexiconInput";
const _GLO = "GetLexiconOutput";
const _GSST = "GetSpeechSynthesisTask";
const _GSSTI = "GetSpeechSynthesisTaskInput";
const _GSSTO = "GetSpeechSynthesisTaskOutput";
const _I = "Id";
const _IALC = "IncludeAdditionalLanguageCodes";
const _ILE = "InvalidLexiconException";
const _INTE = "InvalidNextTokenException";
const _ISBE = "InvalidS3BucketException";
const _ISE = "InvalidSsmlException";
const _ISKE = "InvalidS3KeyException";
const _ISRE = "InvalidSampleRateException";
const _ISTAE = "InvalidSnsTopicArnException";
const _ITIE = "InvalidTaskIdException";
const _L = "Lexicon";
const _LA = "LexiconAttributes";
const _LAe = "LexiconArn";
const _LC = "LexiconContent";
const _LCa = "LanguageCode";
const _LCe = "LexemesCount";
const _LD = "LexiconDescription";
const _LDL = "LexiconDescriptionList";
const _LL = "ListLexicons";
const _LLI = "ListLexiconsInput";
const _LLO = "ListLexiconsOutput";
const _LM = "LastModified";
const _LN = "LexiconNames";
const _LNFE = "LexiconNotFoundException";
const _LNSE = "LanguageNotSupportedException";
const _LNa = "LanguageName";
const _LSEE = "LexiconSizeExceededException";
const _LSST = "ListSpeechSynthesisTasks";
const _LSSTI = "ListSpeechSynthesisTasksInput";
const _LSSTO = "ListSpeechSynthesisTasksOutput";
const _Le = "Lexicons";
const _MLLEE = "MaxLexemeLengthExceededException";
const _MLNEE = "MaxLexiconsNumberExceededException";
const _MNSFFE = "MarksNotSupportedForFormatException";
const _MR = "MaxResults";
const _N = "Name";
const _NT = "NextToken";
const _OF = "OutputFormat";
const _OSBN = "OutputS3BucketName";
const _OSKP = "OutputS3KeyPrefix";
const _OU = "OutputUri";
const _PL = "PutLexicon";
const _PLI = "PutLexiconInput";
const _PLO = "PutLexiconOutput";
const _RC = "RequestCharacters";
const _S = "Size";
const _SE = "SupportedEngines";
const _SFE = "ServiceFailureException";
const _SMNSFTTE = "SsmlMarksNotSupportedForTextTypeException";
const _SMT = "SpeechMarkTypes";
const _SR = "SampleRate";
const _SS = "SynthesizeSpeech";
const _SSI = "SynthesizeSpeechInput";
const _SSO = "SynthesizeSpeechOutput";
const _SSST = "StartSpeechSynthesisTask";
const _SSSTI = "StartSpeechSynthesisTaskInput";
const _SSSTO = "StartSpeechSynthesisTaskOutput";
const _ST = "SynthesisTask";
const _STA = "SnsTopicArn";
const _STNFE = "SynthesisTaskNotFoundException";
const _STy = "SynthesisTasks";
const _St = "Status";
const _T = "Text";
const _TI = "TaskId";
const _TLEE = "TextLengthExceededException";
const _TS = "TaskStatus";
const _TSR = "TaskStatusReason";
const _TT = "TextType";
const _UPAE = "UnsupportedPlsAlphabetException";
const _UPLE = "UnsupportedPlsLanguageException";
const _V = "Voices";
const _VI = "VoiceId";
const _VL = "VoiceList";
const _Vo = "Voice";
const _c = "client";
const _e = "error";
const _h = "http";
const _hE = "httpError";
const _hH = "httpHeader";
const _hQ = "httpQuery";
const _m = "message";
const _s = "streaming";
const _se = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.polly";
const _xaR = "x-amzn-RequestCharacters";
const n0 = "com.amazonaws.polly";
var AudioStream = [0, n0, _AS, { [_s]: 1 }, 42];
var LexiconContent = [0, n0, _LC, 8, 0];
var DeleteLexiconInput$ = [3, n0, _DLI,
    0,
    [_N],
    [[0, 1]], 1
];
var DeleteLexiconOutput$ = [3, n0, _DLO,
    0,
    [],
    []
];
var DescribeVoicesInput$ = [3, n0, _DVI,
    0,
    [_E, _LCa, _IALC, _NT],
    [[0, { [_hQ]: _E }], [0, { [_hQ]: _LCa }], [2, { [_hQ]: _IALC }], [0, { [_hQ]: _NT }]]
];
var DescribeVoicesOutput$ = [3, n0, _DVO,
    0,
    [_V, _NT],
    [() => VoiceList, 0]
];
var EngineNotSupportedException$ = [-3, n0, _ENSE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(EngineNotSupportedException$, EngineNotSupportedException);
var GetLexiconInput$ = [3, n0, _GLI,
    0,
    [_N],
    [[0, 1]], 1
];
var GetLexiconOutput$ = [3, n0, _GLO,
    0,
    [_L, _LA],
    [[() => Lexicon$, 0], () => LexiconAttributes$]
];
var GetSpeechSynthesisTaskInput$ = [3, n0, _GSSTI,
    0,
    [_TI],
    [[0, 1]], 1
];
var GetSpeechSynthesisTaskOutput$ = [3, n0, _GSSTO,
    0,
    [_ST],
    [() => SynthesisTask$]
];
var InvalidLexiconException$ = [-3, n0, _ILE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidLexiconException$, InvalidLexiconException);
var InvalidNextTokenException$ = [-3, n0, _INTE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidNextTokenException$, InvalidNextTokenException);
var InvalidS3BucketException$ = [-3, n0, _ISBE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidS3BucketException$, InvalidS3BucketException);
var InvalidS3KeyException$ = [-3, n0, _ISKE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidS3KeyException$, InvalidS3KeyException);
var InvalidSampleRateException$ = [-3, n0, _ISRE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidSampleRateException$, InvalidSampleRateException);
var InvalidSnsTopicArnException$ = [-3, n0, _ISTAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidSnsTopicArnException$, InvalidSnsTopicArnException);
var InvalidSsmlException$ = [-3, n0, _ISE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidSsmlException$, InvalidSsmlException);
var InvalidTaskIdException$ = [-3, n0, _ITIE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InvalidTaskIdException$, InvalidTaskIdException);
var LanguageNotSupportedException$ = [-3, n0, _LNSE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(LanguageNotSupportedException$, LanguageNotSupportedException);
var Lexicon$ = [3, n0, _L,
    0,
    [_C, _N],
    [[() => LexiconContent, 0], 0]
];
var LexiconAttributes$ = [3, n0, _LA,
    0,
    [_A, _LCa, _LM, _LAe, _LCe, _S],
    [0, 0, 4, 0, 1, 1]
];
var LexiconDescription$ = [3, n0, _LD,
    0,
    [_N, _At],
    [0, () => LexiconAttributes$]
];
var LexiconNotFoundException$ = [-3, n0, _LNFE,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(LexiconNotFoundException$, LexiconNotFoundException);
var LexiconSizeExceededException$ = [-3, n0, _LSEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(LexiconSizeExceededException$, LexiconSizeExceededException);
var ListLexiconsInput$ = [3, n0, _LLI,
    0,
    [_NT],
    [[0, { [_hQ]: _NT }]]
];
var ListLexiconsOutput$ = [3, n0, _LLO,
    0,
    [_Le, _NT],
    [() => LexiconDescriptionList, 0]
];
var ListSpeechSynthesisTasksInput$ = [3, n0, _LSSTI,
    0,
    [_MR, _NT, _St],
    [[1, { [_hQ]: _MR }], [0, { [_hQ]: _NT }], [0, { [_hQ]: _St }]]
];
var ListSpeechSynthesisTasksOutput$ = [3, n0, _LSSTO,
    0,
    [_NT, _STy],
    [0, () => SynthesisTasks]
];
var MarksNotSupportedForFormatException$ = [-3, n0, _MNSFFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(MarksNotSupportedForFormatException$, MarksNotSupportedForFormatException);
var MaxLexemeLengthExceededException$ = [-3, n0, _MLLEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(MaxLexemeLengthExceededException$, MaxLexemeLengthExceededException);
var MaxLexiconsNumberExceededException$ = [-3, n0, _MLNEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(MaxLexiconsNumberExceededException$, MaxLexiconsNumberExceededException);
var PutLexiconInput$ = [3, n0, _PLI,
    0,
    [_N, _C],
    [[0, 1], [() => LexiconContent, 0]], 2
];
var PutLexiconOutput$ = [3, n0, _PLO,
    0,
    [],
    []
];
var ServiceFailureException$ = [-3, n0, _SFE,
    { [_e]: _se, [_hE]: 500 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ServiceFailureException$, ServiceFailureException);
var SsmlMarksNotSupportedForTextTypeException$ = [-3, n0, _SMNSFTTE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(SsmlMarksNotSupportedForTextTypeException$, SsmlMarksNotSupportedForTextTypeException);
var StartSpeechSynthesisTaskInput$ = [3, n0, _SSSTI,
    0,
    [_OF, _OSBN, _T, _VI, _E, _LCa, _LN, _OSKP, _SR, _STA, _SMT, _TT],
    [0, 0, 0, 0, 0, 0, 64 | 0, 0, 0, 0, 64 | 0, 0], 4
];
var StartSpeechSynthesisTaskOutput$ = [3, n0, _SSSTO,
    0,
    [_ST],
    [() => SynthesisTask$]
];
var SynthesisTask$ = [3, n0, _ST,
    0,
    [_E, _TI, _TS, _TSR, _OU, _CT, _RC, _STA, _LN, _OF, _SR, _SMT, _TT, _VI, _LCa],
    [0, 0, 0, 0, 0, 4, 1, 0, 64 | 0, 0, 0, 64 | 0, 0, 0, 0]
];
var SynthesisTaskNotFoundException$ = [-3, n0, _STNFE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(SynthesisTaskNotFoundException$, SynthesisTaskNotFoundException);
var SynthesizeSpeechInput$ = [3, n0, _SSI,
    0,
    [_OF, _T, _VI, _E, _LCa, _LN, _SR, _SMT, _TT],
    [0, 0, 0, 0, 0, 64 | 0, 0, 64 | 0, 0], 3
];
var SynthesizeSpeechOutput$ = [3, n0, _SSO,
    0,
    [_AS, _CTo, _RC],
    [[() => AudioStream, 16], [0, { [_hH]: _CT_ }], [1, { [_hH]: _xaR }]]
];
var TextLengthExceededException$ = [-3, n0, _TLEE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(TextLengthExceededException$, TextLengthExceededException);
var UnsupportedPlsAlphabetException$ = [-3, n0, _UPAE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(UnsupportedPlsAlphabetException$, UnsupportedPlsAlphabetException);
var UnsupportedPlsLanguageException$ = [-3, n0, _UPLE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(UnsupportedPlsLanguageException$, UnsupportedPlsLanguageException);
var Voice$ = [3, n0, _Vo,
    0,
    [_G, _I, _LCa, _LNa, _N, _ALC, _SE],
    [0, 0, 0, 0, 0, 64 | 0, 64 | 0]
];
var PollyServiceException$ = [-3, _sm, "PollyServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(PollyServiceException$, PollyServiceException);
var LexiconDescriptionList = [1, n0, _LDL,
    0, () => LexiconDescription$
];
var SynthesisTasks = [1, n0, _STy,
    0, () => SynthesisTask$
];
var VoiceList = [1, n0, _VL,
    0, () => Voice$
];
var DeleteLexicon$ = [9, n0, _DL,
    { [_h]: ["DELETE", "/v1/lexicons/{Name}", 200] }, () => DeleteLexiconInput$, () => DeleteLexiconOutput$
];
var DescribeVoices$ = [9, n0, _DV,
    { [_h]: ["GET", "/v1/voices", 200] }, () => DescribeVoicesInput$, () => DescribeVoicesOutput$
];
var GetLexicon$ = [9, n0, _GL,
    { [_h]: ["GET", "/v1/lexicons/{Name}", 200] }, () => GetLexiconInput$, () => GetLexiconOutput$
];
var GetSpeechSynthesisTask$ = [9, n0, _GSST,
    { [_h]: ["GET", "/v1/synthesisTasks/{TaskId}", 200] }, () => GetSpeechSynthesisTaskInput$, () => GetSpeechSynthesisTaskOutput$
];
var ListLexicons$ = [9, n0, _LL,
    { [_h]: ["GET", "/v1/lexicons", 200] }, () => ListLexiconsInput$, () => ListLexiconsOutput$
];
var ListSpeechSynthesisTasks$ = [9, n0, _LSST,
    { [_h]: ["GET", "/v1/synthesisTasks", 200] }, () => ListSpeechSynthesisTasksInput$, () => ListSpeechSynthesisTasksOutput$
];
var PutLexicon$ = [9, n0, _PL,
    { [_h]: ["PUT", "/v1/lexicons/{Name}", 200] }, () => PutLexiconInput$, () => PutLexiconOutput$
];
var StartSpeechSynthesisTask$ = [9, n0, _SSST,
    { [_h]: ["POST", "/v1/synthesisTasks", 200] }, () => StartSpeechSynthesisTaskInput$, () => StartSpeechSynthesisTaskOutput$
];
var SynthesizeSpeech$ = [9, n0, _SS,
    { [_h]: ["POST", "/v1/speech", 200] }, () => SynthesizeSpeechInput$, () => SynthesizeSpeechOutput$
];

class DeleteLexiconCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "DeleteLexicon", {})
    .n("PollyClient", "DeleteLexiconCommand")
    .sc(DeleteLexicon$)
    .build() {
}

class DescribeVoicesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "DescribeVoices", {})
    .n("PollyClient", "DescribeVoicesCommand")
    .sc(DescribeVoices$)
    .build() {
}

class GetLexiconCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "GetLexicon", {})
    .n("PollyClient", "GetLexiconCommand")
    .sc(GetLexicon$)
    .build() {
}

class GetSpeechSynthesisTaskCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "GetSpeechSynthesisTask", {})
    .n("PollyClient", "GetSpeechSynthesisTaskCommand")
    .sc(GetSpeechSynthesisTask$)
    .build() {
}

class ListLexiconsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "ListLexicons", {})
    .n("PollyClient", "ListLexiconsCommand")
    .sc(ListLexicons$)
    .build() {
}

class ListSpeechSynthesisTasksCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "ListSpeechSynthesisTasks", {})
    .n("PollyClient", "ListSpeechSynthesisTasksCommand")
    .sc(ListSpeechSynthesisTasks$)
    .build() {
}

class PutLexiconCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "PutLexicon", {})
    .n("PollyClient", "PutLexiconCommand")
    .sc(PutLexicon$)
    .build() {
}

class StartSpeechSynthesisTaskCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "StartSpeechSynthesisTask", {})
    .n("PollyClient", "StartSpeechSynthesisTaskCommand")
    .sc(StartSpeechSynthesisTask$)
    .build() {
}

class SynthesizeSpeechCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "SynthesizeSpeech", {})
    .n("PollyClient", "SynthesizeSpeechCommand")
    .sc(SynthesizeSpeech$)
    .build() {
}

const paginateListSpeechSynthesisTasks = core.createPaginator(PollyClient, ListSpeechSynthesisTasksCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    DeleteLexiconCommand,
    DescribeVoicesCommand,
    GetLexiconCommand,
    GetSpeechSynthesisTaskCommand,
    ListLexiconsCommand,
    ListSpeechSynthesisTasksCommand,
    PutLexiconCommand,
    StartSpeechSynthesisTaskCommand,
    SynthesizeSpeechCommand,
};
const paginators = {
    paginateListSpeechSynthesisTasks,
};
class Polly extends PollyClient {
}
smithyClient.createAggregatedClient(commands, Polly, { paginators });

const Engine = {
    GENERATIVE: "generative",
    LONG_FORM: "long-form",
    NEURAL: "neural",
    STANDARD: "standard",
};
const LanguageCode = {
    ar_AE: "ar-AE",
    arb: "arb",
    ca_ES: "ca-ES",
    cmn_CN: "cmn-CN",
    cs_CZ: "cs-CZ",
    cy_GB: "cy-GB",
    da_DK: "da-DK",
    de_AT: "de-AT",
    de_CH: "de-CH",
    de_DE: "de-DE",
    en_AU: "en-AU",
    en_GB: "en-GB",
    en_GB_WLS: "en-GB-WLS",
    en_IE: "en-IE",
    en_IN: "en-IN",
    en_NZ: "en-NZ",
    en_SG: "en-SG",
    en_US: "en-US",
    en_ZA: "en-ZA",
    es_ES: "es-ES",
    es_MX: "es-MX",
    es_US: "es-US",
    fi_FI: "fi-FI",
    fr_BE: "fr-BE",
    fr_CA: "fr-CA",
    fr_FR: "fr-FR",
    hi_IN: "hi-IN",
    is_IS: "is-IS",
    it_IT: "it-IT",
    ja_JP: "ja-JP",
    ko_KR: "ko-KR",
    nb_NO: "nb-NO",
    nl_BE: "nl-BE",
    nl_NL: "nl-NL",
    pl_PL: "pl-PL",
    pt_BR: "pt-BR",
    pt_PT: "pt-PT",
    ro_RO: "ro-RO",
    ru_RU: "ru-RU",
    sv_SE: "sv-SE",
    tr_TR: "tr-TR",
    yue_CN: "yue-CN",
};
const Gender = {
    Female: "Female",
    Male: "Male",
};
const VoiceId = {
    Aditi: "Aditi",
    Adriano: "Adriano",
    Amy: "Amy",
    Andres: "Andres",
    Aria: "Aria",
    Arlet: "Arlet",
    Arthur: "Arthur",
    Astrid: "Astrid",
    Ayanda: "Ayanda",
    Bianca: "Bianca",
    Brian: "Brian",
    Burcu: "Burcu",
    Camila: "Camila",
    Carla: "Carla",
    Carmen: "Carmen",
    Celine: "Celine",
    Chantal: "Chantal",
    Conchita: "Conchita",
    Cristiano: "Cristiano",
    Daniel: "Daniel",
    Danielle: "Danielle",
    Dora: "Dora",
    Elin: "Elin",
    Emma: "Emma",
    Enrique: "Enrique",
    Ewa: "Ewa",
    Filiz: "Filiz",
    Gabrielle: "Gabrielle",
    Geraint: "Geraint",
    Giorgio: "Giorgio",
    Gregory: "Gregory",
    Gwyneth: "Gwyneth",
    Hala: "Hala",
    Hannah: "Hannah",
    Hans: "Hans",
    Hiujin: "Hiujin",
    Ida: "Ida",
    Ines: "Ines",
    Isabelle: "Isabelle",
    Ivy: "Ivy",
    Jacek: "Jacek",
    Jan: "Jan",
    Jasmine: "Jasmine",
    Jihye: "Jihye",
    Jitka: "Jitka",
    Joanna: "Joanna",
    Joey: "Joey",
    Justin: "Justin",
    Kajal: "Kajal",
    Karl: "Karl",
    Kazuha: "Kazuha",
    Kendra: "Kendra",
    Kevin: "Kevin",
    Kimberly: "Kimberly",
    Laura: "Laura",
    Lea: "Lea",
    Liam: "Liam",
    Lisa: "Lisa",
    Liv: "Liv",
    Lotte: "Lotte",
    Lucia: "Lucia",
    Lupe: "Lupe",
    Mads: "Mads",
    Maja: "Maja",
    Marlene: "Marlene",
    Mathieu: "Mathieu",
    Matthew: "Matthew",
    Maxim: "Maxim",
    Mia: "Mia",
    Miguel: "Miguel",
    Mizuki: "Mizuki",
    Naja: "Naja",
    Niamh: "Niamh",
    Nicole: "Nicole",
    Ola: "Ola",
    Olivia: "Olivia",
    Pedro: "Pedro",
    Penelope: "Penelope",
    Raveena: "Raveena",
    Remi: "Remi",
    Ricardo: "Ricardo",
    Ruben: "Ruben",
    Russell: "Russell",
    Ruth: "Ruth",
    Sabrina: "Sabrina",
    Salli: "Salli",
    Seoyeon: "Seoyeon",
    Sergio: "Sergio",
    Sofie: "Sofie",
    Stephen: "Stephen",
    Suvi: "Suvi",
    Takumi: "Takumi",
    Tatyana: "Tatyana",
    Thiago: "Thiago",
    Tomoko: "Tomoko",
    Vicki: "Vicki",
    Vitoria: "Vitoria",
    Zayd: "Zayd",
    Zeina: "Zeina",
    Zhiyu: "Zhiyu",
};
const OutputFormat = {
    JSON: "json",
    MP3: "mp3",
    OGG_OPUS: "ogg_opus",
    OGG_VORBIS: "ogg_vorbis",
    PCM: "pcm",
};
const SpeechMarkType = {
    SENTENCE: "sentence",
    SSML: "ssml",
    VISEME: "viseme",
    WORD: "word",
};
const TaskStatus = {
    COMPLETED: "completed",
    FAILED: "failed",
    IN_PROGRESS: "inProgress",
    SCHEDULED: "scheduled",
};
const TextType = {
    SSML: "ssml",
    TEXT: "text",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.DeleteLexicon$ = DeleteLexicon$;
exports.DeleteLexiconCommand = DeleteLexiconCommand;
exports.DeleteLexiconInput$ = DeleteLexiconInput$;
exports.DeleteLexiconOutput$ = DeleteLexiconOutput$;
exports.DescribeVoices$ = DescribeVoices$;
exports.DescribeVoicesCommand = DescribeVoicesCommand;
exports.DescribeVoicesInput$ = DescribeVoicesInput$;
exports.DescribeVoicesOutput$ = DescribeVoicesOutput$;
exports.Engine = Engine;
exports.EngineNotSupportedException = EngineNotSupportedException;
exports.EngineNotSupportedException$ = EngineNotSupportedException$;
exports.Gender = Gender;
exports.GetLexicon$ = GetLexicon$;
exports.GetLexiconCommand = GetLexiconCommand;
exports.GetLexiconInput$ = GetLexiconInput$;
exports.GetLexiconOutput$ = GetLexiconOutput$;
exports.GetSpeechSynthesisTask$ = GetSpeechSynthesisTask$;
exports.GetSpeechSynthesisTaskCommand = GetSpeechSynthesisTaskCommand;
exports.GetSpeechSynthesisTaskInput$ = GetSpeechSynthesisTaskInput$;
exports.GetSpeechSynthesisTaskOutput$ = GetSpeechSynthesisTaskOutput$;
exports.InvalidLexiconException = InvalidLexiconException;
exports.InvalidLexiconException$ = InvalidLexiconException$;
exports.InvalidNextTokenException = InvalidNextTokenException;
exports.InvalidNextTokenException$ = InvalidNextTokenException$;
exports.InvalidS3BucketException = InvalidS3BucketException;
exports.InvalidS3BucketException$ = InvalidS3BucketException$;
exports.InvalidS3KeyException = InvalidS3KeyException;
exports.InvalidS3KeyException$ = InvalidS3KeyException$;
exports.InvalidSampleRateException = InvalidSampleRateException;
exports.InvalidSampleRateException$ = InvalidSampleRateException$;
exports.InvalidSnsTopicArnException = InvalidSnsTopicArnException;
exports.InvalidSnsTopicArnException$ = InvalidSnsTopicArnException$;
exports.InvalidSsmlException = InvalidSsmlException;
exports.InvalidSsmlException$ = InvalidSsmlException$;
exports.InvalidTaskIdException = InvalidTaskIdException;
exports.InvalidTaskIdException$ = InvalidTaskIdException$;
exports.LanguageCode = LanguageCode;
exports.LanguageNotSupportedException = LanguageNotSupportedException;
exports.LanguageNotSupportedException$ = LanguageNotSupportedException$;
exports.Lexicon$ = Lexicon$;
exports.LexiconAttributes$ = LexiconAttributes$;
exports.LexiconDescription$ = LexiconDescription$;
exports.LexiconNotFoundException = LexiconNotFoundException;
exports.LexiconNotFoundException$ = LexiconNotFoundException$;
exports.LexiconSizeExceededException = LexiconSizeExceededException;
exports.LexiconSizeExceededException$ = LexiconSizeExceededException$;
exports.ListLexicons$ = ListLexicons$;
exports.ListLexiconsCommand = ListLexiconsCommand;
exports.ListLexiconsInput$ = ListLexiconsInput$;
exports.ListLexiconsOutput$ = ListLexiconsOutput$;
exports.ListSpeechSynthesisTasks$ = ListSpeechSynthesisTasks$;
exports.ListSpeechSynthesisTasksCommand = ListSpeechSynthesisTasksCommand;
exports.ListSpeechSynthesisTasksInput$ = ListSpeechSynthesisTasksInput$;
exports.ListSpeechSynthesisTasksOutput$ = ListSpeechSynthesisTasksOutput$;
exports.MarksNotSupportedForFormatException = MarksNotSupportedForFormatException;
exports.MarksNotSupportedForFormatException$ = MarksNotSupportedForFormatException$;
exports.MaxLexemeLengthExceededException = MaxLexemeLengthExceededException;
exports.MaxLexemeLengthExceededException$ = MaxLexemeLengthExceededException$;
exports.MaxLexiconsNumberExceededException = MaxLexiconsNumberExceededException;
exports.MaxLexiconsNumberExceededException$ = MaxLexiconsNumberExceededException$;
exports.OutputFormat = OutputFormat;
exports.Polly = Polly;
exports.PollyClient = PollyClient;
exports.PollyServiceException = PollyServiceException;
exports.PollyServiceException$ = PollyServiceException$;
exports.PutLexicon$ = PutLexicon$;
exports.PutLexiconCommand = PutLexiconCommand;
exports.PutLexiconInput$ = PutLexiconInput$;
exports.PutLexiconOutput$ = PutLexiconOutput$;
exports.ServiceFailureException = ServiceFailureException;
exports.ServiceFailureException$ = ServiceFailureException$;
exports.SpeechMarkType = SpeechMarkType;
exports.SsmlMarksNotSupportedForTextTypeException = SsmlMarksNotSupportedForTextTypeException;
exports.SsmlMarksNotSupportedForTextTypeException$ = SsmlMarksNotSupportedForTextTypeException$;
exports.StartSpeechSynthesisTask$ = StartSpeechSynthesisTask$;
exports.StartSpeechSynthesisTaskCommand = StartSpeechSynthesisTaskCommand;
exports.StartSpeechSynthesisTaskInput$ = StartSpeechSynthesisTaskInput$;
exports.StartSpeechSynthesisTaskOutput$ = StartSpeechSynthesisTaskOutput$;
exports.SynthesisTask$ = SynthesisTask$;
exports.SynthesisTaskNotFoundException = SynthesisTaskNotFoundException;
exports.SynthesisTaskNotFoundException$ = SynthesisTaskNotFoundException$;
exports.SynthesizeSpeech$ = SynthesizeSpeech$;
exports.SynthesizeSpeechCommand = SynthesizeSpeechCommand;
exports.SynthesizeSpeechInput$ = SynthesizeSpeechInput$;
exports.SynthesizeSpeechOutput$ = SynthesizeSpeechOutput$;
exports.TaskStatus = TaskStatus;
exports.TextLengthExceededException = TextLengthExceededException;
exports.TextLengthExceededException$ = TextLengthExceededException$;
exports.TextType = TextType;
exports.UnsupportedPlsAlphabetException = UnsupportedPlsAlphabetException;
exports.UnsupportedPlsAlphabetException$ = UnsupportedPlsAlphabetException$;
exports.UnsupportedPlsLanguageException = UnsupportedPlsLanguageException;
exports.UnsupportedPlsLanguageException$ = UnsupportedPlsLanguageException$;
exports.Voice$ = Voice$;
exports.VoiceId = VoiceId;
exports.paginateListSpeechSynthesisTasks = paginateListSpeechSynthesisTasks;
